### PR TITLE
Use static lambda keyword

### PR DESCRIPTION
**Pull request by iSazonov**
# PR Summary

Use static lambda keyword where possible to make code readability better.

**No functional changes.**
**No changes in generated IL code** (one exclusion - one delegate was converted to static).

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [ ] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

**Review by vexx32 — APPROVED** [review]
I'm not super familiar with this feature, but from what I've read of it and what I can see here, this looks like a good change to make. Nice work!

**Comment by iSazonov** [comment]
> I'm not super familiar with this feature,

In general, Lambda does two allocations - (1) a delegate, (2) an instance of temporary class.
If a lambda uses only parameter arguments in its body Roslyn excludes second allocation (no closure created).
Roslyn already does the optimization for years. Now C# allows to add "static" keyword for such lambdas. This explicitly shows that the lambda does not use "external" variables - this makes a code readability better, also this protects from a regression (is the lambda is on hot path adding "external" variable can raise huge allocations.).

**Review by TravisEz13 — APPROVED** [review]

**Comment by TravisEz13** [comment]
@PoshChan Please remind me in 1 day

**Comment by PoshChan** [comment]
@TravisEz13, this is the reminder you requested 1 day ago

**Comment by TravisEz13** [comment]
@PoshChan Please remind me in 30 minutes

**Comment by PoshChan** [comment]
@TravisEz13, this is the reminder you requested 30 minutes ago

File: src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs [hunk 1]
```diff
@@ -5822,7 +5822,7 @@ private static TypeCompletionMapping[][] InitializeTypeCache()
 
             #endregion Process_LoadedAssemblies
 
-            var grouping = entries.Values.GroupBy(t => t.Key.Count(c => c == '.')).OrderBy(g => g.Key).ToArray();
+            var grouping = entries.Values.GroupBy(static t => t.Key.Count(c => c == '.')).OrderBy(static g => g.Key).ToArray();
```
Comment (xtqqczze) [inline src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs:5825]: missed one :p

**Comment by iSazonov** [comment]
It is ask for Roslyn repo. (I asked in .Net Runtime without answer.)

**Comment by xtqqczze** [comment]
> It is ask for Roslyn repo. (I asked in .Net Runtime without answer.)

I can't find an existing issue.

**Comment by iSazonov** [comment]
It was my comment in an PR. :-)

**Comment by xtqqczze** [comment]
@xtqqczze I have submitted a new analyzer proposal: https://github.com/dotnet/roslyn-analyzers/issues/5027.

**Pull request merged**